body.tex

```diff
@@ -1,166 +1,466 @@
+The data repository, including a POSIX datastore and a SQLite registry, was entirely located on EFS, which was mouted to each of the EC2 compute instances.
 
-The papers should cite standard LSST references\footnote{See \url{https://github.com/lsst-pst/LSSTreferences}}, 
-where appropriate. For the usage, please see below.  These examples all use the ADS handle, unless they are 
-project docs then the use the project handle like LSE-17.
+Subsequently, support for the S3 datastore eliminated the need of EFS, and the PostgreSQL RDS replaced the SQLite registry.
+Also, a Personal HTCondor Pool on an single EC2 instance was used before we tested the capability to launch new EC2 instances as the HTCondor workers and execute jobs.
+This approach allowed us to debug and adapt more smoothly and always had a fallback option in integrating and testing new features and components.
 
-All are on the lsst-texmf which you can get from \url{http://lsst-texmf.lsst.io}
+\subsection{Test minimally before scaling up}
 
+Small tasks are tested before workflows of jobs, and small workflows are tested before large workflows.
+The canonical \href{https://github.com/lsst/ci\_hsc}{ci\_hsc} dataset and the accompanying CI workflow provide a minimal HSC test dataset and a representative DRP workflow and algorithms.
+This test workflow is an important step in integrating new changes.
+For larger scale tests we used the \href{https://jira.lsstcorp.org/browse/DM-11345}{HSC-RC2 dataset}.
 
-\subsubsection{LSST System and Science}
+\subsection{Use Generation 3 Middleware}
 
-The LSST system (brief overview of telescope, camera and data management subsystems),
-science drivers and science forecasts are described in:
+One of the mandates in the \poc~was to use the Generation 3 Middleware (\citeds{DMTN-056}), which has been designed to ease the DRP execution and automation compared to the previous Generation 2 Middleware. Generation 3 Middleware design is more readily extensible to alternative backends, compared to Generation 2 Middleware.
+As it was still relatively early in its implementation, findings in the \poc~were fed back to the Generation 3 Middleware team and influenced its development.
 
-\begin{itemize}
-\item LSST Science Requirements Document: \cite{LPM-17}.
-\item LSST overview paper: \cite{2008arXiv0805.2366I}.
-\item LSST Science Book: \cite{abell2009lsst}.
-\end{itemize}
-%------------------------------------------------------------------------------
+On the other hand, as during the \poc~the Generation 3 Middleware was under active development, backwards incompatible changes occurred often, and the APIs were unstable.
+Furthermore, there were no integration tests when the \poc~started and the test coverage has been suboptimal.
+Hence, changes often broke functionality.
+Lack of a good integration test is still a major concern going forward.
+New integration tests may not be accepted due to the much longer time required to run the test suite.
```

Comment:
I would delete this.  It doesn't take any longer if we can run in parallel.  It's a matter of resources.